cmake/CompilerOptions.cmake

```diff
@@ -12,19 +12,6 @@ if(CMAKE_CXX_COMPILER_LOADED)
              ${CMAKE_CXX_COMPILER_VERSION}")
 endif()
 
-# Workaround the missing support of shared_ptr for arrays.
-
-check_cxx_feature(HAVE_SHARED_PTR_ARRAY_EXTENSION
-                  shared_ptr_array_extension.cpp
-                  "has C++17 extension of shared_ptr for arrays")
```

Comment:
I vaguely recall that this test failed on macOS without `boost`. Do you know if the test was just broken/incorrect somehow?

You should be able to remove the file `./cmake/utils/cxx_tests/shared_ptr_array_extension.cpp` now I think

Reply:
... and actually I guess a lot of those could maybe be deleted now that we trimmed our set of `check_cxx_feature` tests

Reply:
> I vaguely recall that this test failed on macOS without `boost`. Do you know if the test was just
> broken/incorrect somehow?

Yes it was circa 2019 for MacOS on travis, which were probably lagging behind in terms of OS release.... 
That's when I added the workaround.
I checked the configure section of all builds (except wheels which I still do not understand quite well and for which I do not see where is the configuration step) and for all of them the check_cxx_feature passed, i.e. all compilers we use have the feature now.
 
> You should be able to remove the file `./cmake/utils/cxx_tests/shared_ptr_array_extension.cpp` now I think
Indeed.

Reply:
> ... and actually I guess a lot of those could maybe be deleted now that we trimmed our set of `check_cxx_feature` tests

Indeed too.

Reply:
Click details on the macos cibuildwheel run to get to 

https://github.com/openmeeg/openmeeg/runs/7670824167?check_suite_focus=true

Then go to the "run cibuildwheel" step, then the library build you can see if you hit/uncollapse "› Running before_all...":

https://github.com/openmeeg/openmeeg/runs/7670824167?check_suite_focus=true#step:4:450

It doesn't say anything about shared pointers but I guess it works! The wheels are tested using `pytest`.